src/common/Compiler.h

```diff
@@ -223,6 +210,26 @@ inline int CountTrailingZeroes(unsigned long long x) { int i = 0; while (i < 64
 #   define ALIGN_STACK_FOR_MINGW
 #endif
 
+/* The noexcept keyword should be used on all move constructors and move
+assignments so that containers move objects instead of copying them.
+The noexcept keyword was added in C++11 but the __cpp_noexcept_function_type
+definition to detect its implementation was only added in C++17. */
+#if defined(__cpp_noexcept_function_type) || __cplusplus >= 201103 
```

Comment:
We should nuke all this NOEXCEPT and CONSTEXPR stuff. It was used because some compilers took many years to finish implementing C++11, but all the supported compilers should be fine now.

Reply:
We better do this in a later PR, especially if we want to rewrite all the occurrences making use of them.

Reply:
Sure, that can be done later, and for now you can remove the commit putting lipstick on the stuff to be nuked :stuck_out_tongue:

Reply:
That code is not only putting lipstick on it, it is moving it to another place in a way it is not defined per compiler. All the compilers run this unique code, which was not true before, it's now deduplicated. I needed this to test other parts of the code with various compiler options.

Basically instead of doing:

```c++
#if GCC
// GCC compiler thing
// C++ language thing
#elif MSVC
// MSVC compiler thing
// C++ language thing
#else
// other compiler thing
// old C++ language thing
#endif
```

We now do:

```c++
#if GCC
// GCC compiler thing
#elif MSVC
// MSVC compiler thing
#else
// other compiler thing
#endif

#if CPLUSPLUSVERSION
// C++ language thing
#else
// old C++ language thing
#endif
```

Reply:
> That code is not only putting lipstick on it, it is moving it to another place in a way it is not defined per compiler.

There's nothing there to cause problems with this. All the definitions of NOEXCEPT* are identical and can simply be deduplicated. All the CONSTEXPR* stuff is already outside of compiler-specific blocks.

Reply:
Deduplicated.